src/client/configuration.cc

```diff
@@ -104,10 +104,12 @@ Configuration::Configuration() {
     List<String> path_dirs;
     base::SplitString<':'>(path, path_dirs);
 
+    String self_path;
+    CHECK(base::GetSelfPath(self_path, nullptr));
```

Comment:
We should print a warning that we can't reliably detect our-self in `$PATH`. And remove `CHECK`. Also we can abort.